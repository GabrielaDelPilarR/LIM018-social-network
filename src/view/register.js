import { createNewUser, observer } from '../firebase/auth.js';

const register = {
  template: () => {
    const register = ` 
    <section class="firstView">
    <div class="containerIndex">
      <img src="./img/cubiertos.png" alt="Cubiertos">
      <h1> HELP TASTER </h1>
      <h2> Registrate aquí</h2>
      <span> Podras ver las fotos de los lugares que tienen la mejor comida de tú país..</span>
      <form class="formRegister">
        <input name="email" type="email" class="inputRegister" placeholder="E-mail">
        <input name="password" type="password" class ="inputRegister" placeholder="Contraseña">
        <button type="submit" class="buttonRegister"> Registrar </button>
      </form>
      <span> Al registrarte, aceptas nuestras Condiciones,la Política de datos y la Política de cookies.</span>
    </div>

    <div class="modalContainer">
      <div class="modal">
        <img src="./img/verificacion.webp"></img> 
        <span>Para continuar con la creación de tu nueva cuenta de Help Taster, verifica tu dirección de correo electrónico.</span>
      </div>
    </div>
    </section>`;

    const divRegister = document.createElement('div');
    divRegister.classList.add('registers');
    divRegister.innerHTML = register;
    return divRegister;
  },

  init: () => {
    const formRegister = document.querySelector('.formRegister');

    formRegister.addEventListener('submit', (event) => {
      event.preventDefault();
      const form = new FormData(event.target);
      // createNewUser(form.get('email'), form.get('password')).then(
      //   const modal = document.querySelector('.modalContainer');
      //   modal.style.visibility = 'visible';
      // ).catch();
      createNewUser(form.get('email'), form.get('password'));
      const modal = document.querySelector('.modalContainer');
      modal.style.visibility = 'visible';
    });
  },
};

export default register;
