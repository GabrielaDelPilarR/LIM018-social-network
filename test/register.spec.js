import register from '../src/view/register';

jest.mock('../src/firebase/auth.js', () => ({
  loginGoogle: jest.fn(),
  createNewUser: jest.fn(),

}));

describe('login', () => {
  it('should be a function', () => {
    expect(typeof register.template).toBe('function');
    expect(typeof register.init).toBe('function');
  });

  it('should check template', () => {
    expect(register.template()).toMatchSnapshot();
  });

  it('should render template login', () => {
    document.body.appendChild(register.template());
    register.init();
    const formEmail = document.querySelector('.form-email');
    formEmail.submit();
    expect(formEmail instanceof HTMLElement).toBeTruthy();
  });
});
