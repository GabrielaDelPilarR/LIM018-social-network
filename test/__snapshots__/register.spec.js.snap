// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register should check template 1`] = `
<div
  class="registers"
>
   
    
  <section
    class="first-view"
  >
    
      
    <div
      class="container-index"
    >
      
        
      <img
        alt="Cubiertos"
        src="./img/cubiertos.png"
      />
      
        
      <h1>
         HELP TASTER 
      </h1>
      
        
      <h2>
         Registrate aquí
      </h2>
      
        
      <span>
         Podrás ver las fotos de los lugares que tienen la mejor comida de tú país..
      </span>
      
        
      <form
        class="form-email"
      >
        
          
        <input
          class="input-register"
          id="inputName"
          name="name"
          placeholder="Nombre "
          type="text"
        />
        
          
        <input
          class="input-register"
          id="inputEmail"
          name="email"
          placeholder="E-mail"
          type="email"
        />
        
          
        <div>
          
            
          <input
            class="input-register"
            id="inputPassword"
            name="password"
            placeholder="Contraseña"
            type="password"
          />
          
            
          <p
            class="condicion"
          >
             Tu contraseña debe contener mínimo 6 carácteres
          </p>
          
          
        </div>
        
          
        <button
          class="button-register"
          type="submit"
        >
           Registrar 
        </button>
        
        
      </form>
      
        
      <span
        class="message-error"
      >
        Llena los campos
      </span>
      
        
      <div
        class="separator"
      >
        <hr
          class="hr"
        />
        O
        <hr
          class="hr"
        />
      </div>
      
        
      <div
        class="button-index"
      >
        
          
        <button
          class="button"
          id="googleLogin"
        >
          
            
          <img
            src="./img/Google.png"
          />
          
            Inicia con Google
          
        </button>
        
        
      </div>
      
        
      <span>
         Al registrarte, aceptas nuestras Condiciones,la Política de datos y la Política de cookies.
      </span>
      
      
    </div>
    
      
    <div
      class="modal-container"
    >
       
        
      <div
        class="modal"
      >
        
          
        <img
          src="./img/verificacion.webp"
        />
         
          
        <span>
          Para continuar con la creación de tu nueva cuenta de Help Taster, verifica tu dirección de correo electrónico.
        </span>
        
          
        <button
          class="btn-modal"
        >
           Aceptar 
        </button>
        
        
      </div>
      
      
    </div>
    
    
  </section>
</div>
`;
